src/data/parts.js

```diff
@@ -0,0 +1,82 @@
+ import { parts, cars } from "./init.js";
+import { ObjectId } from "mongodb";
+import Validator from "../utils/validator.js";
+import { DataBaseException, NotFoundException } from "../utils/exceptions.js";
+
+const createPart = async (name, price, manufacturer, sellerId, carIds) => {
+    const partData = Validator.validatePart({
+        name,
+        price,
+        manufacturer,
+        sellerId,
+        carIds
+    });
+
+    let newPart = {
+        name: partData.name,
+        price: partData.price,
+        manufacturer: partData.manufacturer,
+        sellerId: new ObjectId(partData.sellerId),
+        carIds: partData.carIds.map(id => new ObjectId(id)),
+        dateAdded: new Date()
+    };
+
+    try {
+        const partsCollection = await parts();
+        const insertInfo = await partsCollection.insertOne(newPart);
+        if (!insertInfo.acknowledged || !insertInfo.insertedId) {
+            throw new Error('Could not add part');
+        }
+        const newId = insertInfo.insertedId;
+        return await getPartById(newId.toString());
+    } catch (e) {
+        throw new DataBaseException(e.message);
+    }
+};
+
+const getPartById = async (partId) => {
+    const validPartId = Validator.validateId(partId);
+    try {
+        const partsCollection = await parts();
+        const part = await partsCollection.findOne({ _id: new ObjectId(validPartId) });
+        if (!part) throw new NotFoundException('Part not found');
+        return part;
+    } catch (e) {
+        throw new DataBaseException(e.message);
+    }
+};
+
+const searchPartsByName = async (searchQuery) => {
+    searchQuery = Validator.validateString(searchQuery, 'Search Query');
+    try {
+        const partsCollection = await parts();
+        const parts = await partsCollection.find(
+            { name: { $regex: searchQuery, $options: 'i' } },

```

Comment:
I would use or to have the search query match manufacture and car make and model as well.
example:
const results = await partsCollection.aggregate([
    // Step 1: Join with the cars collection
    {
        $lookup: {
            from: "cars",                // The collection to join with
            localField: "manufacturer",  // Field from partsCollection
            foreignField: "make",        // Field from cars collection
            as: "carDetails"             // Alias for the joined data
        }
    },
    // Step 2: Unwind the joined data (optional, depending on structure)
    {
        $unwind: {
            path: "$carDetails",
            preserveNullAndEmptyArrays: true // Optional: If you want parts with no car match
        }
    },
    // Step 3: Filter based on the search query across multiple fields
    {
        $match: {
            $or: [
                { name: { $regex: searchQuery, $options: 'i' } },
                { manufacturer: { $regex: searchQuery, $options: 'i' } },
                { "carDetails.make": { $regex: searchQuery, $options: 'i' } },
                { "carDetails.model": { $regex: searchQuery, $options: 'i' } }
            ]
        }
    }
]).toArray();